002_simulation_structure/002_simulation_structure.tex

```diff
@@ -1,14 +1,112 @@
+
+\begin{figure}[htb]
+    \centering
+    \includegraphics[width=0.3\linewidth]{002_simulation_structure/images/health_global.pdf}
+    \caption{The health of the agents is represented by a HP value between \texttt{HPCritical} and \texttt{MaxHP}. All HP values which are below \texttt{WeakLevel} are included in the critical state. The diagram is not drawn to scale.}
+    \label{fig:health_system}
+\end{figure}
+
+%
+%%%%%%%% Food and Health: updateHP
+%
+\subsection{Food and Health: \texttt{updateHP}}\label{updateHP}
+To upgrade their HP, the agents need to eat. However, a certain saturation in the HP gain occurs when the agents keep eating. Moreover, we consider diminishing returns in terms of HP change as a function of the food eaten. Mathematically, the ideas of diminishing returns and saturation are well captured by the step response of a 1st order system:
+
+\begin{equation}\label{updateHP_general}
+   \text{newHP}= \text{currentHP} +\underbrace{w\cdot(1-e^{-\text{foodTaken}/ \tau})}_{=\text{HPChange}}
+\end{equation}
+
+where the two parameters $w$ and $\tau$ are defined at the beginning of the simulation. The shape of this curve is given in \Cref{fig:updateHP} together with some important parameters.
+
+\begin{figure}[htb]%
+    \centering
+    \subfloat[\centering Overview]{{\includegraphics[width=0.36\linewidth]{002_simulation_structure/images/health_updateHP_overview.pdf}}}%
+    \qquad
+    \subfloat[\centering Detailed representation]{{\includegraphics[width=0.36\linewidth]{002_simulation_structure/images/health_updateHP_detailed.pdf}}}%
+    \caption{\texttt{updateHP} as a function of the amount of food eaten ("FoodTaken").}%
+    \label{fig:updateHP}%
+\end{figure}
+
+Note that it is not possible to gain more HP than $w$ over the duration of one day. Indeed, we want to prevent the agents health to change too quickly. As an example, we can think of an agent that starts from the weak level and wants to reach the maximum HP value. It would take several days for this agent to "recover" from this weak level and stabilizes its health to a large HP value. Also, it is possible for an agent to get an HP value that is larger than \texttt{maxHP} with this function. However, the obtained HP value will be reduced by the cost of living using the function \texttt{hpDecay} (see below). After this operation, the final HP value will be bounded by \texttt{maxHP}.
```

Comment:
```suggestion
It is not possible to gain more HP than $w$ over the duration of one day; this is an intentional limit to prevent an agent's health from improving too quickly. As an example, we can think of an agent that starts from the weak level and wants to reach the maximum HP value. It would take several days for this agent to ``recover'' from this weak level and stabilise its health to a high HP value. 

Note that it is possible for an agent to achieve an HP value that is larger than \texttt{MaxHP} inside \lstinline$hpDecay()$. At the end of each day, the \lstinline$hpDecay()$ function will apply the cost of living and then bound the final HP value by \lstinline$MaxHP$.
```